server/src/test/java/org/elasticsearch/cluster/coordination/PersistedStateIT.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.cluster.coordination;
+
+import org.elasticsearch.action.admin.cluster.settings.ClusterUpdateSettingsRequest;
+import org.elasticsearch.action.admin.cluster.state.ClusterStateRequest;
+import org.elasticsearch.cluster.metadata.MetaData;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.test.ESIntegTestCase;
+import org.elasticsearch.test.discovery.TestZenDiscovery;
+
+import java.util.concurrent.TimeUnit;
+
+import static org.hamcrest.Matchers.equalTo;
+
+@ESIntegTestCase.ClusterScope(scope = ESIntegTestCase.Scope.TEST, numDataNodes = 0)
+public class PersistedStateIT extends ESIntegTestCase {
+
+    private static Settings SETTINGS = Settings.builder()
+        .put(TestZenDiscovery.USE_ZEN2.getKey(), true)
+        .put(TestZenDiscovery.USE_ZEN2_PERSISTED_STATE.getKey(), true)
+        .build();
+
+
+    public void testPersistentSettingsOnFullRestart() throws Exception {
```

Comment:
Please see my point 5 above. I was trying to test IndexMetaData as well, but this requires proper recovery implementation. Also, this test passes with InMemoryPersistedState, because of dandling indices. I'll add this test once recovery is properly implemented.